### PR TITLE
#1658 Allowed finance team to edit all RR's

### LINKED ISSUE #1658: Finance Fixes - Finance Team should be able to edit all RR's
### Description

Right now only the creator of a reimbursement request is able to edit it. We want to allow any member of the finance team to edit any reimbursement request as well.

### Acceptance Criteria

- [ ] Backend edit reimbursement request endpoint is updated so that finance members have permissions
- [ ] Frontend edit reimbursement request check that enables/disables the edit button on an RR is updated so that finance members have permissions.

### Proposed Solution
In the frontend and backend we check in other endpoints/ places whether a user is on the finance team, and enable/disable things based on that. You should be able to follow those examples.

### PR DESCRIPTION
**Pull request by jessicayzhao**
## Changes

Allowed any member of the finance team to edit RR's.

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1658

### CODE REVIEW

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 1]
```diff
@@ -255,10 +255,12 @@ export default class ReimbursementRequestService {
...
     if (oldReimbursementRequest.dateDeleted) throw new DeletedException('Reimbursement Request', requestId);
-    if (oldReimbursementRequest.recipientId !== submitter.userId)
-      throw new AccessDeniedException(
-        'You do not have access to delete this reimbursement request, only the creator can edit a reimbursement request'
-      );
+    try {
+      await validateUserIsPartOfFinanceTeam(submitter);
+    } catch {
+      if (oldReimbursementRequest.recipientId !== submitter.userId)
+        throw new AccessDeniedException('You do not have access to edit this reimbursement request');
```
Comment (callmejim1226) [inline src/backend/src/services/reimbursement-requests.services.ts:262]: It would be better if we specifically mention that only the creator of RR or a member of the finance team can edit RR here as an error msg.

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 2]
```diff
@@ -255,10 +255,12 @@ export default class ReimbursementRequestService {
 
     if (!oldReimbursementRequest) throw new NotFoundException('Reimbursement Request', requestId);
     if (oldReimbursementRequest.dateDeleted) throw new DeletedException('Reimbursement Request', requestId);
-    if (oldReimbursementRequest.recipientId !== submitter.userId)
-      throw new AccessDeniedException(
-        'You do not have access to delete this reimbursement request, only the creator can edit a reimbursement request'
-      );
+    try {
```
Comment (callmejim1226) [inline src/backend/src/services/reimbursement-requests.services.ts:258]: I see your point of using try and catch here to give a specific error message in case, the submitter is not a finance team member. However, when we look at how we used `validateUserIsPartOfFinanceTeam` in other RR service functions, we let the error be thrown from the `validateUserIsPartOfFinanceTeam` function if the submitter of edit RR is not a member of the finance team.

This is a way to keep our service function code simple and easier to understand. It also makes sense to give a general error msg of `You are not a member of the finance team!` in the case, that the submitter of edit RR is not a member of the finance team. 

In other words, we just want to use `validateUserIsPartOfFinanceTeam` function here instead of adding try and catch. We can keep the check if the submitter is a creator after `validateUserIsPartOfFinanceTeam` function usage.

File: src/backend/tests/reimbursement-requests.test.ts [hunk 3]
```diff
@@ -222,11 +222,24 @@ describe('Reimbursement Requests', () => {
           [],
           superman
         )
-      ).rejects.toThrow(
-        new AccessDeniedException(
-          'You do not have access to delete this reimbursement request, only the creator can edit a reimbursement request'
+      ).rejects.toThrow(new AccessDeniedException('You do not have access to edit this reimbursement request'));
```
Comment (callmejim1226) [inline src/backend/tests/reimbursement-requests.test.ts:225]: Let's update this error msg after we resolve the error msg from the service function.

File: src/backend/tests/reimbursement-requests.test.ts [hunk 4]
```diff
@@ -222,11 +222,24 @@ describe('Reimbursement Requests', () => {
...
+          GiveMeMyMoney.vendorId,
+          GiveMeMyMoney.account as ClubAccount,
+          GiveMeMyMoney.expenseTypeId,
+          GiveMeMyMoney.totalCost,
+          [],
+          [],
+          alfred
         )
-      );
+      ).rejects.toThrow(new AccessDeniedException('You do not have access to edit this reimbursement request'));
```
Comment (callmejim1226) [inline src/backend/tests/reimbursement-requests.test.ts:242]: Let's update this error msg after we change the error msg from the service function.

**Review by callmejim1226 — CHANGES_REQUESTED** [review]
Nice start on this ticket!!

I left a few comments on the error message in the RR edit service function.

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 5]
```diff
@@ -255,10 +255,14 @@ export default class ReimbursementRequestService {
...
+      await validateUserIsPartOfFinanceTeam(submitter);
+    } catch {
+      if (oldReimbursementRequest.recipientId !== submitter.userId)
+        throw new AccessDeniedException(
+          'You do not have access to edit this reimbursement request, only the creator or finance team can edit a reimbursement request'
```
Comment (callmejim1226) [inline src/backend/src/services/reimbursement-requests.services.ts:263]: Let's shrink the error message to say `Only the creator or finance team can edit a reimbursement request` since we know this error is the access denied type.

**Review by callmejim1226 — APPROVED** [review]
Looks good to me. Good job!

File: src/backend/src/services/reimbursement-requests.services.ts (hunk 2 above)
Comment (RChandler234) [inline src/backend/src/services/reimbursement-requests.services.ts:258]: this logic is right, but bc it's a bit weird, we should abstract this try/catch to a helper method in reimbursement request utils

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/backend/src/utils/reimbursement-requests.utils.ts [hunk 6]
```diff
@@ -233,3 +233,12 @@ export const isAuthUserHeadOfFinance = (user: Prisma.UserGetPayload<typeof authU
 const isTeamIdInList = (teamId: string, teamsList: Team[]) => {
   return teamsList.map((team) => team.teamId).includes(teamId);
 };
+
+export const validateUserEditReimbursementRequest = async (user: User, reimbursementRequest: Reimbursement_Request) => {
```
Comment (RChandler234) [inline src/backend/src/utils/reimbursement-requests.utils.ts:237]: can you rename this to validateUserEditRRPermissions and then add a comment about what it's doing, like the comment for createNewProducts

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥
